cumulus/polkadot-parachain/src/service.rs

```diff
@@ -521,119 +428,171 @@ where
+		client: Arc<ParachainClient<RuntimeApi>>,
+		block_import: ParachainBlockImport<RuntimeApi>,
+		config: &Configuration,
+		telemetry_handle: Option<TelemetryHandle>,
+		task_manager: &TaskManager,
+	) -> sc_service::error::Result<DefaultImportQueue<Block>> {
+		let verifier_client = client.clone();
+
+		let aura_verifier =
+			cumulus_client_consensus_aura::build_verifier::<<AuraId as AppCrypto>::Pair, _, _, _>(
+				cumulus_client_consensus_aura::BuildVerifierParams {
+					client: verifier_client.clone(),
+					create_inherent_data_providers: move |parent_hash, _| {
+						let cidp_client = verifier_client.clone();
+						async move {
+							let slot_duration = cumulus_client_consensus_aura::slot_duration_at(
+								&*cidp_client,
+								parent_hash,
+							)?;
+							let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+
+							let slot =
+						sp_consensus_aura::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
+							*timestamp,
+							slot_duration,
+						);
+
+							Ok((slot, timestamp))
+						}
+					},
+					telemetry: telemetry_handle,
+				},
+			);
 
-				Ok((slot, timestamp))
-			}
-		},
-		telemetry: telemetry_handle,
-	});
-
-	let relay_chain_verifier =
-		Box::new(RelayChainVerifier::new(client.clone(), |_, _| async { Ok(()) })) as Box<_>;
-
-	let verifier = Verifier {
-		client,
-		relay_chain_verifier,
-		aura_verifier: Box::new(aura_verifier),
-		_phantom: PhantomData,
-	};
+		let relay_chain_verifier =
+			Box::new(RelayChainVerifier::new(client.clone(), |_, _| async { Ok(()) }));
+
+		let verifier = Verifier {
+			client,
+			relay_chain_verifier,
+			aura_verifier: Box::new(aura_verifier),
+			_phantom: PhantomData,
+		};
 
-	let registry = config.prometheus_registry();
-	let spawner = task_manager.spawn_essential_handle();
+		let registry = config.prometheus_registry();
+		let spawner = task_manager.spawn_essential_handle();
 
-	Ok(BasicQueue::new(verifier, Box::new(block_import), None, &spawner, registry))
+		Ok(BasicQueue::new(verifier, Box::new(block_import), None, &spawner, registry))
+	}
 }
 
 /// Uses the lookahead collator to support async backing.
 ///
 /// Start an aura powered parachain node. Some system chains use this.
-pub async fn start_generic_aura_lookahead_node<Net: NetworkBackend<Block, Hash>>(
-	parachain_config: Configuration,
-	polkadot_config: Configuration,
-	collator_options: CollatorOptions,
-	para_id: ParaId,
-	hwbench: Option<sc_sysinfo::HwBench>,
-) -> sc_service::error::Result<(TaskManager, Arc<ParachainClient<FakeRuntimeApi>>)> {
-	start_node_impl::<FakeRuntimeApi, _, _, _, Net>(
-		parachain_config,
-		polkadot_config,
-		collator_options,
-		CollatorSybilResistance::Resistant, // Aura
-		para_id,
-		build_parachain_rpc_extensions::<FakeRuntimeApi>,
-		build_relay_to_aura_import_queue::<_, AuraId>,
-		start_lookahead_aura_consensus::<_, AuraId>,
-		hwbench,
-	)
-	.await
+pub(crate) struct GenericAuraLookaheadNode<RuntimeApi, AuraId>(
```

Comment:
Not in this PR, but in a bain-dead follow-up, breaking this file into smaller ones might also be good, to yield a quick overview of all options that is available.

Reply:
Yes, this is just a quick clean-up where I tried to address some low hanging fruits. There are still a lot of things to do. Some of the things on my list:
- making the `NodeSpec` more generic and moving it under the `common` module. The same for `BuildImportQueue`, etc
- maybe splitting the part related to running the commands in a separate trait from `DynNodeSpec` and also separate file.
- splitting the implementations of `BuildImportQueue`, `StartConsensus` in separate files and also moving them under `common`
- also I would like to reduce the number of arguments for `StartConsensus`. Not sure if possible and not sure how yet, but now it takes 15 arguments which is a lot. Maybe grouping them somehow.